core/runtime/tests/error.rs

```diff
@@ -76,37 +76,32 @@ if (errMessage !== "higher-level sync error: original sync error") {
   throw new Error("unexpected error message from op_err_sync: " + errMessage);
 }
 "#,
-      )
-      .unwrap();
-
-    let promise = runtime
-      .execute_script_static(
-        "test_error_context_async.js",
-        r#"
+    )
+    .unwrap();
 
+  let promise = runtime
+    .execute_script_static(
+      "test_error_context_async.js",
+      r#"
 (async () => {
-let errMessage;
-try {
-  await Deno.core.opAsync("op_err_async");
-} catch (err) {
-  errMessage = err.message;
-}
-if (errMessage !== "higher-level async error: original async error") {
-  throw new Error("unexpected error message from op_err_async: " + errMessage);
-}
+  let errMessage;
+  try {
+    await Deno.core.opAsync("op_err_async");
+  } catch (err) {
+    errMessage = err.message;
+  }
+  if (errMessage !== "higher-level async error: original async error") {
+    throw new Error("unexpected error message from op_err_async: " + errMessage);
+  }
 })()
 "#,
-      )
-      .unwrap();
+    ).unwrap();
 
-    match runtime.poll_value(cx, &promise) {
-      Poll::Ready(Ok(_)) => {}
-      Poll::Ready(Err(err)) => panic!("{err:?}"),
-      _ => panic!(),
-    }
-    Poll::Ready(())
-  })
-  .await;
+  let resolve = runtime.resolve(promise);
+  futures::executor::block_on(
+    runtime.with_event_loop_promise(resolve, PollEventLoopOptions::default()),
+  )
+  .unwrap();
```

Comment:
Since it's an async test fn, then maybe `.await` it instead?

Reply:
Good call.. I unwrapped the inner poll_fn later on and missed this one.